boards/arm/b_g474e_dpow1/board.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/adc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(board, LOG_LEVEL_DBG);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(usbc_port)
+
+#define VBUS DT_PATH(vbus)
+
+/* Common settings supported by most ADCs */
+
+static int16_t sample_buffer;
+
+static const uint32_t output_ohm = DT_PROP(VBUS, output_ohms);
+static const uint32_t full_ohm = DT_PROP(VBUS, full_ohms);
+
+/* Data of ADC io-channels specified in devicetree. */
+static const struct adc_dt_spec adc_channel = ADC_DT_SPEC_GET(VBUS);
+
+static struct adc_sequence sequence = {
+	.channels = BIT(adc_channel.channel_id),
+	.buffer = &sample_buffer,
+	/* buffer size in bytes, not number of samples */
+	.buffer_size = sizeof(sample_buffer),
+	.resolution = adc_channel.resolution,
+};
```

Comment:
note: there's `adc_sequence_init_dt` to populate certain fields from DT.

Reply:
Done